include/dsn/internal/thrift_parser.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * The MIT License (MIT)
+ *
```

Comment:
let's move this file to src/core/tools/common with http_parsers.h/.cpp etc.  parsers are plugged and do not need to be exposed to others.